spec/Section 5 -- Validation.md

```diff
@@ -429,10 +429,10 @@ SameResponseShape(fieldA, fieldB):
 
   * Let {typeA} be the return type of {fieldA}.
   * Let {typeB} be the return type of {fieldB}.
-  * If {typeA} or {typeB} is Non-Null.
-    * If {typeA} or {typeB} is nullable, return false.
-    * Let {typeA} be the nullable type of {typeA}
-    * Let {typeB} be the nullable type of {typeB}
+  * If {typeA} is Non-Null.
+    * Let {typeA} be the unwrapped nullable type of {typeA}
+  * If {typeB} is Non-Null.
+    * Let {typeB} be the unwrapped nullable type of {typeB}
```

Comment:
is this sufficient? what about more complex covariance scenarios?

Reply:
I think this is what we need to compare against. Two valid branches into https://spec.graphql.org/draft/#IsValidImplementationFieldType()

Reply:
I think that is correct: we would have to consider the same logic as linked above.